hw/ip/otbn/dv/rig/rig/gens/bad_ispr.py

```diff
@@ -103,4 +103,11 @@ def _gen(self,
             assert imm_idx is not None
             prog_insn.operands[imm_idx] = bad_addr
 
+            # UNIMP instruction is equivalent to CSRRW x0, 0xC00, x0
+            # In the case of a randomly generated CSRRW, it is a good idea to
+            # have this instruction most of the time.
+            if random.random() < 0.8:
+                if self.insns[idx].mnemonic == "csrrw":
+                    prog_insn.operands = [0, 0xC00, 0]
```

Comment:
I understand what this code does and I think it is correct, but I don't understand *why* this code exists in the first place. (You write "it is a good idea", but why?)

I suggest to modify/extend the comment to describe that. For example:
```suggestion
            # If this is a CSRRW, randomly turn it into an UNIMP instruction, which is equivalent to CSRRW x0, 0xC00, x0.
            # This substantially increases the coverage of the UNIMP instruction.
            if self.insns[idx].mnemonic == "csrrw" and random.random() < 0.8:
                prog_insn.operands = [0, 0xC00, 0]
```
The suggestion also un-nests the `if` statements, which I don't see the need for. (Formatting may be rubbish since I wrote it in the GitHub comment editor. :slightly_smiling_face: )

Reply:
Thanks for the suggestion @andreaskurth! I think you're right, I should have been more clear. The need comes from this conversion: https://github.com/lowRISC/opentitan/pull/7923#pullrequestreview-739636528. The file that this PR changes is for generating read/writes for bad CSR indices. So everything was already there, this just tweaks it a little bit to make it more likely to see `UNIMP` specifically.